web/src/components/MapView.tsx

```diff
@@ -113,40 +102,58 @@ const MapView = React.forwardRef((props: MapViewProps, ref: React.Ref<MapRef>):
     }
   }, [currentFeature, featureCollection, queryLocation, queryParams, selectFeature])
 
-  const onSelectFeature = (e: MapEvent) => {
-    if (e.features?.length) {
-      flyToPoi(e.features[0].geometry.coordinates)
-      selectFeature(e.features[0])
-      changeSnapPoint(1)
-      queryParams.set(locationName, e.features[0].properties.urlSlug)
-      updateQueryParams(queryParams)
-    } else {
-      setQueryLocation(null)
-      selectFeature(null)
-      updateQueryParams()
-    }
-  }
+  const onDeselectFeature = useCallback(() => {
+    setQueryLocation(null)
+    selectFeature(null)
+    updateQueryParams()
+  }, [selectFeature, setQueryLocation])
+
+  const onSelectFeature = useCallback(
+    event => {
+      // Since the event is only fired if canvas on layer was clicked, the container propagation has to be stopped on deselect
+      event.originalEvent.stopPropagation()
```

Comment:
the container fires an (deselection) event to the child f.e. the canvas layer where the pois are placed. this should not be propagated because there is a onClick Select event on it. If i would not stop propagation both events would fire

Reply:
Thank you for explaining.

Reply:
gonna add in 902